### PR TITLE
update integration test to use fractions and dispatch events asynchronously

### PR DESCRIPTION
**Pull request by benouaer**

### CODE REVIEW

File: framework-tools-fraction/pom.xml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <parent>
+        <artifactId>framework-tools</artifactId>
+        <groupId>uk.gov.justice</groupId>
+        <version>2.1.0-SNAPSHOT</version>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>framework-tools-fraction</artifactId>
```
Comment (purple52) [inline framework-tools-fraction/pom.xml]: Can we rename this to "archive-loader-fraction"so it describes what it does?

File: framework-tools-fraction/pom.xml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            <groupId>org.wildfly.swarm</groupId>
+            <artifactId>spi</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.wildfly.swarm</groupId>
+            <artifactId>logging</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-log4j12</artifactId>
```
Comment (purple52) [inline framework-tools-fraction/pom.xml]: Why is this needed? We should be using slf4j-api and/or wildfly logging for the fraction if required, unless its for testing only in which case it should be test scope.

File: framework-tools-fraction/pom.xml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            <groupId>uk.gov.justice.services</groupId>
+            <artifactId>event-repository-jdbc</artifactId>
+            <version>${framework.version}</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
```
Comment (purple52) [inline framework-tools-fraction/pom.xml]: Compiler plugin should be configured by the parent of this project, so this config should not be needed.

File: framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/ReplayIntegrationIT.java [hunk 4]
```diff
@@ -1,74 +1,115 @@
...
 import uk.gov.justice.services.eventsourcing.repository.jdbc.exception.InvalidSequenceIdException;
 import uk.gov.justice.services.messaging.JsonEnvelope;
 import uk.gov.justice.services.messaging.Metadata;
 
-import java.io.BufferedReader;
-import java.io.File;
-import java.io.FileFilter;
-import java.io.InputStreamReader;
+import javax.sql.DataSource;
+import java.io.*;
```
Comment (purple52) [inline framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/ReplayIntegrationIT.java]: Please expand wildcard imports as per coding style guidelines

File: pom.xml [hunk 5]
```diff
@@ -16,20 +16,22 @@
...
         <cpp.repo.name>framework-tools</cpp.repo.name>
         <uk.gov.justice.utils.version>1.5.0</uk.gov.justice.utils.version>
+        <version.swarm.fraction-plugin>77</version.swarm.fraction-plugin>
+
         <!--
                 note h2 version is because swarm has its own version of h2 which we must use
                 for the integration tests to work as h2 version inter-compatibility is volatile
         -->
-        <com.h2database.version>1.4.187</com.h2database.version>
+        <com.h2database.version>1.4.187</com.h2database.version><!--1.4.187-->
```
Comment (purple52) [inline pom.xml]: Remove commented out code

File: replay-tool-it/pom.xml [hunk 6]
```diff
@@ -10,12 +10,12 @@
     <modelVersion>4.0.0</modelVersion>
     <packaging>pom</packaging>
     <modules>
-        <module>framework-event-listener</module>
-        <module>framework-tools-it</module>
+        <module>it-example-listener</module>
+        <module>integration-test</module>
```
Comment (purple52) [inline replay-tool-it/pom.xml]: Given that the groupId is very generic (uk.gov.justice), I think these submodule names really need "framework-tools" at the beginning, otherwise we'll have a GAV of uk.gov.justice:integration-test:2.1.0 which will mean nothing to anyone.

Comment (benouaer) [reply replay-tool-it/pom.xml]: agreed
